s3/pkg/datastore/aws/aws.go

```diff
@@ -61,9 +65,49 @@ func (myc *s3Cred) IsExpired() bool {
 	return false
 }
 
+func (ad *AwsAdapter) BucketCreate(ctx context.Context, in *pb.Bucket) error {
+
+	log.Info("Bucket create is called in aws service")
+	log.Info("The input request to create bucket is:", in)
+	Credentials := credentials.NewStaticCredentials(ad.backend.Access, ad.backend.Security, "")
+	configuration := &aws.Config{
+		Region:      aws.String(DefaultRegion),
+		Endpoint:    aws.String(ad.backend.Endpoint),
+		Credentials: Credentials,
+	}
+	svc := awss3.New(session.New(configuration))
+
+	input := &awss3.CreateBucketInput{
+		Bucket: aws.String(in.Name),
+	}
+
+	buckout, err := svc.CreateBucket(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case awss3.ErrCodeBucketAlreadyExists:
+				log.Error(awss3.ErrCodeBucketAlreadyExists, aerr.Error())
+			case awss3.ErrCodeBucketAlreadyOwnedByYou:
+				log.Error(awss3.ErrCodeBucketAlreadyOwnedByYou, aerr.Error())
+			default:
+				log.Error(aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			log.Error(err.Error())
+		}
+		return err
+	}
+	log.Debug("The bucket creation successful in aws s3 service")
+	log.Info("The result of create bucket:", buckout)
+
+	return nil
+}
+
 func (ad *AwsAdapter) Put(ctx context.Context, stream io.Reader, object *pb.Object) (dscommon.PutResult, error) {
-	bucket := ad.backend.BucketName
-	objectId := object.BucketName + "/" + object.ObjectKey
+	bucket := object.BucketName
+	objectId := object.ObjectKey
```

Comment:
nit: objectId  -> objectID

Reply:
I agree, but 'objectId' instead of objectID ...is there in all other codes also, so simply keeping.

Reply:
IMO, We can avoid legacy coding conventions errors